### DISCUSSION TITLE
Requirement issue

### DISCUSSION THREAD

**Discussion by robertnicjoo** [spatie/laravel-sitemap#428] [Q&A]
This package will not install on websites served by `PHP 8.X` it gives following error:

```
Your requirements could not be resolved to an installable set of packages.
```

The reason for error above is that requirement packages for this package still are not supporting `PHP 8.X`. For instance 

```
require spatie/crawler ^5.0 || ^7.0
spatie/browsershot[3.14.0, ..., 3.20.0] require php ^7.0
spatie/image[1.5.3, ..., 1.7.6] require php ^7.0
```

Apparently this package needs to upgrade it's require packages before be ready for `PHP 8`.

**Comment by freekmurze** [comment]
The latest version supports PHP 8: https://github.com/spatie/laravel-sitemap/blob/ae0e7198c5f378f3ab8a4d0ae91891bd00718fad/composer.json#L19

**Reply by robertnicjoo** [reply]
Yes as I said `requirement packages` are not supporting `PHP 8` this package might support but because it's requirements does not therefor it will not install.

**Reply by robertnicjoo** [reply]
you might need to consider updating those packages for php 8 as well.

**Reply by freekmurze** [reply]
All those package support PHP 8. Or maybe I don't understand your issue. Feel free to PR improvements to this or our other packages.

**Reply by cod3rshotout** [reply]
This issue is still there @freekmurze you have to update the external packages otherwise it won't install on php 8:

![Immagine 2022-07-29 145341](https://user-images.githubusercontent.com/72454584/181764719-bc7b1c65-473f-43ae-9067-b3902d072284.jpg)

**Reply by freekmurze** [reply]
Seems like it is just working for me 🤷

<img width="864" alt="CleanShot 2022-07-29 at 17 27 03@2x" src="https://user-images.githubusercontent.com/483853/181792849-5cdee553-3bd4-4580-9ce0-90148901d449.png">

**Reply by prajapatinirav** [reply]
@cod3rshotout  I have the same issue, And after investigation, I found [aws/aws-sdk-php](https://packagist.org/packages/aws/aws-sdk-php) is creating an issue for me.  Update "aws-sdk-php" to latest and it will resolve the issue. Check once with all locked dependencies. In my case, It internally locked the guzzle package which is the root cause of this issue.  
![Screenshot 2022-08-01 at 10 37 14 AM](https://user-images.githubusercontent.com/7949561/182076367-e10392af-7c1d-4a04-94cc-8a43373138e6.png)
@freekmurze
